converter/shieldhit/geo.py

```diff
@@ -0,0 +1,168 @@
+from converter.solid_figures import SolidFigure, BoxFigure, CylinderFigure, SphereFigure
+from abc import ABC
+from dataclasses import dataclass, field
+from math import log10, ceil
+
+
+def format_float(number: float, n: int) -> float:
+    """
+    Format float to be up to n characters wide, as precise as possible and as short
+    as possible (in descending priority). so for example given 12.333 for n=5 you will
+    get 12.33, n=7 will be 12.333
+    """
+    # If number is zero we just want to get 0.0 (it would mess up the log10 operation below)
+    if number == 0:
+        return 0.0
+
+    # Sign messes up the log10 we use do determine how long the number is (we fix that with abs).
+    # Since there is no sign function in python we use `if else`
+    sign = -1 if number < 0 else 1
+    result = float(sign*round(abs(number), n-ceil(log10(number))-1))
```

Comment:
You are still taking `log10(number)` where number may be negative. 
Add good unit tests for negative numbers